torch/utils/data/chunk/__init__.py

```diff
@@ -0,0 +1,46 @@
+from ..dataset import IterableDataset
+from torch._C.data.chunk import ChunkDatasetOptions, SamplerWrapper
+from torch._C.data.chunk import Sampler, RandomSampler, SequentialSampler
+from torch._C.data.chunk import DistributedSampler, DistributedRandomSampler, DistributedSequentialSampler
+from torch._C.data.chunk import ChunkDataReaderUint8T, ChunkDataReaderInt8T, ChunkDataReaderInt16T, ChunkDataReaderInt32T
+from torch._C.data.chunk import ChunkDataReaderInt64T, ChunkDataReaderFloat, ChunkDataReaderDouble
+
+class ChunkDatasetWrapper(IterableDataset):
+    r"""
+    Wrapper class for the C++ ChunkDataset class.
+
+    Additional to the role of wrapping its C++ counterpart dataset class,
+    it also allows conversion or transformation through ``transform_fn``.
+    ``ChunkDatasetWrapper`` extends ``IterableDataset`` because the size of the dataset is
+    unknown.
+
+    Arguments:
+        dataset (Dataset): The whole Dataset
+        transform_fn (optional, callable): Collates, converts or transform batches
+    """
+    def __init__(self, dataset, transform_fn=None):
+        super(ChunkDatasetWrapper, self).__init__()
+        self.dataset = dataset
+        self.transform_fn = transform_fn
+
+    def __iter__(self):
+        return self
+
+    def __len__(self):
+        raise NotImplementedError()
+
+    def __next__(self):
+        batch = self.dataset.get_batch()
+        if batch is None:
+            raise StopIteration
+        if self.transform_fn is not None:
+            return self.transform_fn(batch)
```

Comment:
Why we need the transform function here?

Reply:
We already has a collate function, so I don't think we need transform here.

Reply:
ssnl asked to not use the DataLoader collate_fn, so I moved it inside ChunkDataset and renamed we are not collating (we get a full batch from the library already)